src/ui/common/src/components/pages/workflows/components/CheckItem.tsx

```diff
@@ -26,35 +26,56 @@ export const getCheckStatusIcon = (
   check: CheckPreview,
   tooltipText?: string
 ): JSX.Element => {
-  let statusIcon = (
+  const successIcon = (
     <StatusIndicator
       status={ExecutionStatus.Succeeded}
       tooltipText={tooltipText}
     />
   );
 
+  const errorIcon = (
+    <StatusIndicator
+      status={ExecutionStatus.Failed}
+      tooltipText={tooltipText}
+    />
+  );
+
+  const warningIcon = (
+    <StatusIndicator
+      status={ExecutionStatus.Warning}
+      tooltipText={tooltipText}
+    />
+  );
+
+  let statusIcon = successIcon;
+
   switch (check.status) {
     case ExecutionStatus.Succeeded: {
+      statusIcon = successIcon;
+
       // now we check the value to see if we should show warning or error icon
+      // Can this case even happen, or is the ExecutionState always going to be failed in this case?
+      // I thought ExecutionState of failed meant there was a runtime error ...
+      // May not need this if statement after all.
```

Comment:
Yep if the workflow succeeded an error level check cannot have failed.

Reply:
Sounds good 👌 will update logic here 🧠